delivery_carrier_warehouse/views/stock_warehouse.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="view_warehouse_inherit" model="ir.ui.view">
+        <field name="name">stock.warehouse.inherit</field>
+        <field name="model">stock.warehouse</field>
+        <field name="inherit_id" ref="stock.view_warehouse" />
+        <field name="arch" type="xml">
+            <field name="partner_id" position="after">
+                <field
+                    name="delivery_carrier_id"
+                    domain="[('company_id', 'in', (False, company_id))]"
```

Comment:
Normally, if you use `check_company=True` on the field attribute, the domain is automatically set by the ORM

Reply:
thanks for your knowledges